cores/esp8266/coredecls.h

```diff
@@ -17,14 +17,21 @@ extern bool timeshift64_is_set;
 void esp_yield();
 void esp_schedule();
 void tune_timeshift64 (uint64_t now_us);
-void settimeofday_cb (void (*cb)(void));
 void disable_extra4k_at_link_time (void) __attribute__((noinline));
 
 uint32_t sqrt32 (uint32_t n);
 uint32_t crc32 (const void* data, size_t length, uint32_t crc = 0xffffffff);
 
 #ifdef __cplusplus
 }
+
+#include <functional>
+
+using functionalVoidVoid = std::function<void()>;
```

Comment:
This is rather confusing. 
How about just functionalVoid? Alternatively, a function that takes no args is called a nullary function. A function that doesn't return a value is called a void function. How about:
-nullaryVoid
-functionalNullaryVoid
-functionalVoidNullary
-some variation of the above